### DISCUSSION TITLE
GitHub Container Registry link to a repo

### DISCUSSION THREAD

**Discussion by wei** [community/community#26400] [Packages]
<p>The <a href="https://docs.github.com/en/packages/getting-started-with-github-container-registry/migrating-to-github-container-registry-for-docker-images" rel="nofollow noopener">help page</a> says</p>
<blockquote>
<p>The GitHub Container Registry supersedes the existing Packages Docker registry</p>
</blockquote>
<p>I have two questions regarding Migrating to GitHub Container Registry for Docker images:</p>
<ol>
<li>Can I link a container in the new container registry to a repo, and will it show up under packages? If not currently, what is the roadmap and expected behavior once implemented?</li>
<li>Now that I’m pushing to the new Container Registry, How can I delete a docker image in the old Packages Docker registry to prevent old images being used by mistake? <code>:latest</code> tag</li>
</ol>
<p>Thanks!</p>

**Comment by clarkbw** [comment, marked as answer]
<ol>
<li>Yes! You can link to a repository via the OCI source label, here are 2 ways to do that. (Apologies as I thought this was in the docs)</li>
</ol>
<p>The preferred method is via the Dockerfile like so:</p>
<pre><code class="lang-auto"># repo URL
LABEL org.opencontainers.image.source = "https://github.com/github/super-linter/"
</code></pre>
<p>Alternatively you can add this at build time like so:</p>
<pre><code class="lang-shell">docker build --label "org.opencontainers.image.source=https://github.com/github/super-linter/"
</code></pre>
<ol start="2">
<li>We’re going to turn on delete for the old Docker images soon.  Private images can be deleted immediately, public image deletion is coming. With the new service you’ll always be able to delete images and tags on your own.</li>
</ol>

**Reply by designermonkey** [reply]
I would like to point out that the example shown on the docker web ui for this does not use an equals `=` symbol in the `LABEL` declaration.

**Comment by wei** [comment]
<p>Thanks! I tagged my images now it’s showing up on my user’s <a href="https://github.com/wei?tab=packages" rel="nofollow noopener">Packages tab</a>. and the right repo is being displayed underneath both GCR and GPR versions.<br>
<img src="https://user-images.githubusercontent.com/5880908/181097489-7170413a-8a8a-48db-a12b-4065241ea907.png" alt="image" data-base62-sha1="vEgPpHV2bfNRg4iugCeOjSmUWdx" width="682" height="310"></p>
<p>However, only one being displayed under my repo’s <a href="https://github.com/wei/pull/packages" rel="nofollow noopener">packages page</a> and it shows the GPR image.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/10d2e7a9a0bf3a1d6c20035b1767c39c84e53203.png" data-download-href="/uploads/short-url/2oPv4QkKqUFqDzC9zRcsNMKtXk7.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/5880908/181097490-c5a86bc7-cdde-4af8-b37c-b3b00fb6b832.png" alt="image" data-base62-sha1="2oPv4QkKqUFqDzC9zRcsNMKtXk7" width="690" height="297" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/10d2e7a9a0bf3a1d6c20035b1767c39c84e53203_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">696×300 11.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I presume this won’t be a problem after the public image deletion feature is in : )</p>

**Comment by wei** [comment]
<p>Hi <a class="mention" href="/clarkbw">@clarkbw</a>, When you say “You can link to a repository”, Do you mean the image will show up under the repo’s packages tab?</p>
<p>I published a <a href="https://github.com/orgs/repo-sync/packages/container/github-sync" rel="nofollow noopener">public image</a> for my repo and it’s showing correctly under repository source as follows:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/547be2781dc2854cbd7f242a9343725a12d75306.png" data-download-href="/uploads/short-url/c3nwO2mpZ634I6NDlsj5BcZaMPI.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/5880908/181097492-2ac319f7-b6ac-40a0-9bfc-8acf442a69e8.png" alt="image" data-base62-sha1="c3nwO2mpZ634I6NDlsj5BcZaMPI" width="690" height="226" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/547be2781dc2854cbd7f242a9343725a12d75306_2_690x226.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/547be2781dc2854cbd7f242a9343725a12d75306_2_1035x339.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/547be2781dc2854cbd7f242a9343725a12d75306_2_1380x452.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/547be2781dc2854cbd7f242a9343725a12d75306_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1532×504 42.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>However, it’s not showing under repo’s packages tab or anywhere on the <a href="https://github.com/repo-sync/github-sync" rel="nofollow noopener">repo page</a>. Do we need to do anything else to get it to show up?</p>

**Comment by clarkbw** [comment]
<p><a class="mention" href="/wei">@wei</a> Yes! That’s the correct way to link the repo and the container.  However right now it won’t show up in the repository page view yet.  We’re waiting on some other fixes to go in and then it will start to appear as planned.</p>

**Comment by 0xbuidler** [comment]
<p>Hello !</p>
<p>Is there a way to push images as public directly ? On my end everytime I push an image even linked (as mentioned above) to a public project, it still publish the image as private.</p>

**Comment by clarkbw** [comment]
<p>Not at this time. Images initially start as private and need to be switched. This should only need to be done once.</p>
<p>Can you explain a bit more about how you’re using this?</p>

**Comment by 0xbuidler** [comment]
<p>Well, I have a public repository: <code>https://github.com/MaximeAubanel/GoGame</code></p>
<p>Initially I was releasing packages using <code>docker.pkg.github.com</code>. But I realized I couldn’t pull them without logging. Then I saw the new Docker Container Registry and tried it. It worked and I can pull images anonymously if they are public - but they were not so I had to move them to public manually. Then I came here to figure out how to link this image to a repo because images were not appearing on the project packages profile.</p>
<p>That’s basically it. I’m just thinking it is a bit odd to do it manually - but I guess it’s an upcoming feature. Maybe could be nice to default publish the image as public if it’s linked with a public repository.</p>

**Comment by clarkbw** [comment]
<p>Yeah, its a little bit of a manual process right now.  We’ll optimize it more soon so it’s easier to get the right defaults for new containers.  Thanks for the feedback!</p>

**Comment by aptalca** [comment]
<p>Is there an api call to switch the package from private to public?</p>
<p>We at <a href="https://github.com/linuxserver" rel="noopener nofollow ugc">linuxserver.io</a> are updating our ci to push images to the new container registry but having a lot of issues with this default private marking.</p>
<p>Currently, only the user whose pat was used to upload the image can manage these images/packages. The other users of the organization (including the owners) cannot even see the packages listed as they are not listed in access management. This forces a manual interaction on a single member of the team for each image (we have hundreds) which is not ideal for an org with ci/cd.</p>
<p>Thanks</p>

**Comment by clarkbw** [comment]
<p>Hello <a class="mention" href="/aptalca">@aptalca</a>! Can you email me? clarkbw <span class="mention">@github.com</span> We’ve got a system that solves for this very issue that we’re using internally but its not yet ready to roll out. Happy to have you try it and get your feedback.  Thanks!</p>

**Comment by lukaszstolarczuk** [comment]
<p>I just encountered the same issue. I mean, our org’s admins couldn’t see our docker images on packages page, because these images were pushed by a CI account (a bot). If I didn’t encounter this comment here I just wouldn’t know where to look for.</p>
<p><a class="mention" href="/clarkbw">@clarkbw</a>, shouldn’t it be described in documentation (e.g. <a href="https://docs.github.com/en/free-pro-team@latest/packages/managing-container-images-with-github-container-registry/configuring-access-control-and-visibility-for-container-images#configuring-access-to-container-images-for-an-organization" rel="noopener nofollow ugc">here</a>) or perhaps please put it in <strong>bold</strong>, if the info is there (and I’m blind :wink: ).</p>
<p>Thanks!</p>

**Comment by jef** [comment]
<p>I’ve had no problem linking a repository via the UI, but now I’d like to unlink. Is that a possibility? Either through UI or API?</p>
